scripts/buildkite/golint.sh

```diff
@@ -2,20 +2,32 @@
 
 set -ex
 
+verify_no_files_changed () {
+  # intentionally capture stderr, so status-errors are also PR-failing.
+  # in particular this catches "dubious ownership" failures, which otherwise
+  # do not fail this check and the $() hides the exit code.
+  if [ -n "$(git status --porcelain  2>&1)" ]; then
```

Comment:
I would recommend using `if [[ ... ]]` if you want to stick to non-empty check.
The reason is "[[" makes it Bash operator while "[" in some systems can still be interpreted as binary (you probably don't wont to know this, but `[` is actually executable file symlinked to /bin/test). Which means the output of git command essentially goes there. If it's big enough, then calling '[' fill fail with too-long-argument error-code.

Usually, it's better to avoid passing unbounded output to `-n` and prefer exit-code (or it is the same?) or something like
```
NLINES=$( git status --porcelain 2>&1 | wc -l )
if [[ $NLINES -ne 0 ]]; then ...`

Reply:
The script executes in `/bin/sh` not sure if that is nessesary, but it means that we should not use `[[ ... ]]`. I've changed so we pass the line count instead of the full output as you suggested. 


> (you probably don't wont to know this, but [ is actually executable file symlinked to /bin/test).

I know, such a horrible hack.